samples/matter/common/src/bridge/bridge_manager.cpp

```diff
@@ -60,30 +61,38 @@ CHIP_ERROR BridgeManager::AddBridgedDevices(MatterBridgedDevice *devices[], Brid
 {
 	chip::Optional<uint8_t> indexes[deviceListSize];
 	uint16_t endpoints[deviceListSize];
+	Platform::UniquePtr<BridgedDeviceDataProvider> providerPtr(dataProvider);
 
 	VerifyOrReturnError(devicesPairIndexes, CHIP_ERROR_INTERNAL);
-
-	return AddBridgedDevices(devices, dataProvider, deviceListSize, devicesPairIndexes, endpoints, indexes);
+	CHIP_ERROR err =
+		DoAddBridgedDevices(devices, providerPtr.get(), deviceListSize, devicesPairIndexes, endpoints, indexes);
```

Comment:
nit: I guess `DoAddBridgedDevices` could take `UniquePtr` argument to make it clear that it takes the owhership.